modules/lang-painless/src/test/java/org/elasticsearch/painless/RegexTests.java

```diff
@@ -125,6 +126,51 @@ public void testNamedGroup() {
         assertEquals("o", exec("Matcher m = /(?<first>f)(?<second>o)o/.matcher('foo'); m.find(); return m.namedGroup('second')"));
     }
 
+    public void testMultilineFlag() {
+        assertEquals(false, exec("'a\nb\nc' =~ /^b$/"));
+        assertEquals(true, exec("'a\nb\nc' =~ /^b$/m"));
+    }
```

Comment:
Alternatively I think we should whitelist entire `java.util.regex` api properly with all exceptions, interfaces, etc. Just exclude Pattern.compile! Then tests can simply do stuff like: 
```
assertEquals(Pattern.MULTILINE, exec("/foo/m.flags()"));
```
The constants in that file and helper methods like `split` seem useful. Also `pattern()` seems really useful for tests, especially if we want to test nuances of escaping.

I can do it if you want, i know the whitelisting stuff is not the most fun...